### PR TITLE
YR/convert relationship to use Pydantic/CIAC-9913

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9913

## Description
Converting the 'relationship' from a plain dictionary to a Pedantic object. To avoid having to make adjustments everywhere we use it, once we've gone through the Pedantic object, we convert it back to a dictionary.

### CODE REVIEW

File: demisto_sdk/commands/content_graph/common.py [hunk 1]
```diff
@@ -270,15 +272,27 @@ def by_schema(path: Path) -> "ContentType":
...
+    target_type: Optional[ContentType] = None
+    mandatorily: Optional[bool] = None
+
+
 class Relationships(dict):
     def add(self, relationship: RelationshipType, **kwargs):
         if relationship not in self.keys():
             self.__setitem__(relationship, [])
-        self.__getitem__(relationship).append(kwargs)
+        self.__getitem__(relationship).append(Relationship.parse_obj(kwargs).dict())
```
Comment (ilaner) [inline demisto_sdk/commands/content_graph/common.py:290]: i think you should use `.dict(exclue_none=True)` so null values will not be parsed.

Comment (RosenbergYehuda) [reply demisto_sdk/commands/content_graph/common.py]: FIixed

**Review by JasBeilin — COMMENTED** [review]
Looks good. Lets run the nightly on it before merging.

File: demisto_sdk/commands/content_graph/tests/create_content_graph_test.py [hunk 2]
```diff
@@ -913,3 +914,28 @@ def test_create_content_graph_with_python_version(
...
+        - Verify that the data was added through the Pydantic 'Relationship' object and complies with its specifications:
+            - Convert boolean strings to boolean objects.
+            - Include only properties defined in the Pydantic model.
+            - Exclude properties with a value of None from the object.
+        - Verify that the data was converted back into a dictionary.
+    """
+    relationships = Relationships()
+    relationships.add(
+        RelationshipType.DEPENDS_ON,
+        mandatorily="True",
```
Comment (ilaner) [inline demisto_sdk/commands/content_graph/tests/create_content_graph_test.py:935]: please add more relationships tests. For instance check `true` string and invalid `target_type`

**Review by ilaner — APPROVED** [review]

**Comment by RosenbergYehuda** [comment]
> Looks good. Lets run the nightly on it before merging.

Passed.
https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/pipelines/914630

File: demisto_sdk/commands/content_graph/tests/create_content_graph_test.py (hunk 2 above)
Comment (RosenbergYehuda) [reply demisto_sdk/commands/content_graph/tests/create_content_graph_test.py]: Done

**Pull request merged**